internal/commands/config_pull_policy_test.go

```diff
@@ -0,0 +1,198 @@
+	h "github.com/buildpacks/pack/testhelpers"
+)
+
+func TestConfigPullPolicy(t *testing.T) {
+	color.Disable(true)
+	defer color.Disable(false)
+	spec.Run(t, "ConfigPullPolicyCommand", testConfigPullPolicyCommand, spec.Random(), spec.Report(report.Terminal{}))
+}
+
+func testConfigPullPolicyCommand(t *testing.T, when spec.G, it spec.S) {
+	var (
+		command      *cobra.Command
+		logger       logging.Logger
+		outBuf       bytes.Buffer
+		tempPackHome string
+		configFile   string
+		assert       = h.NewAssertionManager(t)
+		cfg          = config.Config{}
+	)
+
+	it.Before(func() {
+		var err error
+		logger = ilogging.NewLogWithWriters(&outBuf, &outBuf)
+		tempPackHome, err = ioutil.TempDir("", "pack-home")
+		h.AssertNil(t, err)
+		configFile = filepath.Join(tempPackHome, "config.toml")
+
+		command = commands.ConfigPullPolicy(logger, cfg, configFile)
+		command.SetOut(logging.GetWriterForLevel(logger, logging.InfoLevel))
+	})
+
+	it.After(func() {
+		h.AssertNil(t, os.RemoveAll(tempPackHome))
+	})
+
+	when("#ConfigPullPolicy", func() {
+		when("no policy is specified", func() {
+			it("lists default pull policy", func() {
+				command.SetArgs([]string{})
+
+				h.AssertNil(t, command.Execute())
+
+				assert.Contains(outBuf.String(), "always")
+			})
+		})
+
+		when("policy set to always in config", func() {
+			it("lists always as pull policy", func() {
+				cfg.PullPolicy = "always"
+				command = commands.ConfigPullPolicy(logger, cfg, configFile)
+				command.SetArgs([]string{})
+
+				h.AssertNil(t, command.Execute())
+
+				assert.Contains(outBuf.String(), "always")
+			})
+		})
+
+		when("policy set to never in config", func() {
+			it("lists never as pull policy", func() {
+				cfg.PullPolicy = "never"
+				command = commands.ConfigPullPolicy(logger, cfg, configFile)
+				command.SetArgs([]string{})
+
+				h.AssertNil(t, command.Execute())
+
+				assert.Contains(outBuf.String(), "never")
+			})
+		})
+
+		when("policy set to if-not-present in config", func() {
+			it("lists if-not-present as pull policy", func() {
+				cfg.PullPolicy = "if-not-present"
+				command = commands.ConfigPullPolicy(logger, cfg, configFile)
+				command.SetArgs([]string{})
+
+				h.AssertNil(t, command.Execute())
+
+				assert.Contains(outBuf.String(), "if-not-present")
+			})
+		})
+
+		when("policy provided is the same as configured pull policy", func() {
+			it("provides a helpful message", func() {
+				cfg.PullPolicy = "if-not-present"
+				command = commands.ConfigPullPolicy(logger, cfg, configFile)
+				command.SetArgs([]string{"if-not-present"})
+
+				h.AssertNil(t, command.Execute())
+
+				output := outBuf.String()
+				h.AssertEq(t, strings.TrimSpace(output), `Pull policy is already set to if-not-present`)
+			})
+			it("it does not change the configured policy", func() {
+				command = commands.ConfigPullPolicy(logger, cfg, configFile)
+				command.SetArgs([]string{"never"})
+				assert.Succeeds(command.Execute())
+
```

Comment:
I would perhaps read and verify the config file here as well, to make sure that it doesn't need to be run twice in order to set it.